### PR TITLE
feat: add mask and fix resample_cube_spatial

### PR DESCRIPTION
**Pull request by clausmichele**
Overview of this PR:

1. Implementation of the `mask` process, supporting auto resampling (using `resample_cube_spatial`)
2. Tests for `mask`
3. Improvements of `resample_cube_spatial`, fixing https://github.com/Open-EO/openeo-processes-dask/issues/191
4. Tests for `resample_cube_spatial`
5. Now `resample_spatial` is correctly imported, solving this: https://github.com/Open-EO/openeo-processes-dask/pull/103#issuecomment-1802018611
6. Moved `resample_cube_spatial` out of experimental, since it's necessary within the `mask` process.

### CODE REVIEW

**Comment by clausmichele** [comment]
There's something wrong with the code, the commits are messed up. I will fix it.

File: openeo_processes_dask/specs/openeo-processes
Comment (ValentinaHutter) [inline openeo_processes_dask/specs/openeo-processes:1]: I think the changes in the specification are not related to the changes in the implementation, so we probably do not need them here....

**Comment by ValentinaHutter** [comment]
Nice work! As the codecov already suggests, it would be good to have some tests in place! :)

File: openeo_processes_dask/specs/openeo-processes
Comment (clausmichele) [reply openeo_processes_dask/specs/openeo-processes:1]: yes, I don't know why it got into my various PRs sorry. I will fix this.

**Review by GeraldIr — APPROVED** [review]
lgtm, sorry for the delay on approving this, just got back from holidays.

**Pull request merged**